docs/adr/0015-authentication-for-platform.md

```diff
@@ -0,0 +1,69 @@
+# Authentication for all platform environments
+
+* Status: accepted
+* Deciders: durandom
+* Date: 2021-04-26
+
+Technical Story: 
+
+## Context and Problem Statement
+
+The Operate First environments will span multiple data centers and multiple regions. 
+A central authentication system helps identfiying users and system operators without maintaining seperate user registries.
+This ADR is related to platform systems, such as OpenShift, ArgoCD or ACM - for application workloads refer to (0010-common-auth-for-applications)[0010-common-auth-for-applications.md].
+Authentication is also essential for auditing. User management of the applications should be unified so the same user can access all the systems with a single set of credentials. 
+The username should be unique, but also utilized across systems.
+The accepted solution should provide (SSO)[https://en.wikipedia.org/wiki/Single_sign-on], so the user can cary over the identity across different systems.
+
+## Decision Drivers 
+
+1. For the same user a single user identity is provided to all applications.
+2. All users are able to authenticate using the same credentials
+3. Identity provider is responsible for user privacy
+4. Ease of acquiring an account
+5. Publishing data in legal boundaries without worrying about (PII)(https://en.wikipedia.org/wiki/Personal_data)
+
+## Considered Options
+
+1. University Authentication
+2. Google
+3. GitHub
```

Comment:
@larsks IIRC you had some concerns about replacing MOC SSO - could you elaborate here?

Reply:
Uhhh...it's been a while, I think I missed the notification on this one :). I guess my primary concern is that if we move to another authentication source, we're going to invalidate all the current accounts, which are email based. I think moving away from email addresses is probably a good idea, so I'm not arguing against it (I think it will probably simplify many things), but I think it's going to be a disruptive change.

Reply:
OK, thanks. That's an implementation detail I guess. Maybe a soft transition or a grace period where we have both installed.

Reply:
I think we can mitigate the impact to minimal - prepare user group changes before we switch the auth source for OCP etc. We can probably limit the impact to have disruptive effect on Jupyterhub users with running servers. Which is fine if announced beforehand. :slightly_smiling_face: